### PR TITLE
Remove react-test-renderer

### PR DESCRIPTION
**Pull request by ghost**
Contributes to #774

**Overall change:** Remove `react-test-renderer` and  shallow rendering for `psammead-test-helpers`

**Code changes:**

- Remove `shallowRender` and `shouldShallowMatchSnapshot`.
- Update `isNull` and `shouldMatchSnapshot`
- Use `render` from `@testing-library/react`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by thekp** [comment]
Hi @sadickisaac 

I see you are removing the shallowRendering functionality. We currently have an on-going issue to make use of the `psammead-test-helper` in Simorgh, however the shallow rendering is heavily used in Simorgh for the page body tests (i.e `simorgh/src/app/containers/ArticleMain`: https://github.com/bbc/simorgh/blob/latest/src/app/containers/ArticleMain/index.test.jsx#L35-L54)

Do you know if  `@testing-library/react` package has an equivalent "shallow rendering functionality" we can use ?

**Comment by thekp** [comment]
source of the CI issue btw: https://github.com/bbc/psammead/blob/latest/packages/components/psammead-media-player/src/Placeholder/index.test.jsx#L3

**Comment by ghost** [comment]
> Do you know if @testing-library/react package has an equivalent "shallow rendering functionality" we can use ?

`@testing-library/react` does not support shallow rendering. We are trying to move from using shallow rendering. What will happen is, the test depending on `shallowRendering` will be refactored.

**Comment by thekp** [comment]
> @testing-library/react does not support shallow rendering. We are trying to move from using shallow rendering. What will happen is, the test depending on shallowRendering will be refactored.

possibly major refactors to our Simorgh tests 👀(but I suppose it is inevitable since we want to move away from shallowRendering) I'm going to put my running issue on blocked until you get this one done: https://github.com/bbc/simorgh/issues/3264 👍

File: packages/utilities/psammead-test-helpers/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 2.0.0 | [PR#1917](https://github.com/bbc/psammead/pull/1917) Remove react-test-renderer and shallow rendering |
 | 1.3.2 | [PR#1919](https://github.com/bbc/psammead/pull/1919) Use @testing-library-react |
```
Comment (thekp) [inline packages/utilities/psammead-test-helpers/CHANGELOG.md]:
```suggestion
| 1.3.2 | [PR#1919](https://github.com/bbc/psammead/pull/1919) Accidental version bump |
```

**Review by thekp — APPROVED** [review]
This is ok for psammead, as we don't use shallowRender anywhere here.

This issue is for using test-helper in Simorgh: https://github.com/bbc/simorgh/issues/3264

**Review by DenisHdz — APPROVED** [review]

File: packages/utilities/psammead-test-helpers/src/index.js [hunk 2]
```diff
@@ -11,25 +9,10 @@ export const shouldMatchSnapshot = (title, component) => {
...
-    expect(tree).toMatchSnapshot();
-  });
-};
-
 export const isNull = (title, component) => {
   it(title, () => {
-    const tree = renderer.create(component).toJSON();
-    expect(tree).toBeNull();
+    const { container } = render(component);
+    expect(container.firstChild).toBeNull();
```
Comment (dr3) [inline packages/utilities/psammead-test-helpers/src/index.js:15]: Is this the same thing?  is first child always the top level thing? could you have a second child that was not null?

Comment (ghost) [reply packages/utilities/psammead-test-helpers/src/index.js:15]: How would that look like? First child null and second not null. AFAIK `null` is not rendered which makes the second child that was not null to be the firstChild.

Comment (ghost) [reply packages/utilities/psammead-test-helpers/src/index.js:15]: Maybe we could check if the container has children and assert that there are no children rendered.

**Review by dr3 — APPROVED** [review]

**Pull request merged**